engine/class_modules/sc_warrior.cpp

```diff
@@ -2378,7 +2391,22 @@ struct charge_t : public warrior_attack_t
     }
 
     warrior_attack_t::execute();
-
+    
+    if ( p()->legendary.reprisal->ok() )
+    {
+      if ( p()->buff.shield_block->check() )
+      // Even though it isn't mentionned anywhere in spelldata, reprisal only triggers shield block for 4s
+      {
+        p()->buff.shield_block->extend_duration( p(), 4_s ); 
+      }
+      else
+      {
+        p()->buff.shield_block->trigger( 4_s );
+      }
+      p()->buff.revenge->trigger( 1, buff_t::DEFAULT_VALUE(), 1.0 );
```

Comment:
```suggestion
      p()->buff.revenge->trigger();
```
You don't have to specify all these arguments if they're going to be the default ones (1 stack, default value, 100% chance).